RulesEngine/Ruleset.cs

```diff
@@ -81,217 +67,306 @@ public void AddDirectory(string path, string tag = null)
         }
 
         /// <summary>
-        /// Load rules from a file
+        ///     Load rules from a file
         /// </summary>
-        /// <param name="filename">Filename with rules</param>
-        /// <param name="tag">Tag for the rules</param>
-        public void AddFile(string filename, string tag = null)
+        /// <param name="filename"> Filename with rules </param>
+        /// <param name="tag"> Tag for the rules </param>
+        public void AddFile(string filename, string? tag = null)
         {
             if (string.IsNullOrEmpty(filename))
-            {
-                throw new ArgumentException(filename);
-            }
-
-            if (!File.Exists(filename))
-            {
-                throw new FileNotFoundException(filename);
-            }
+                throw new ArgumentException(nameof(filename));
 
             if (_logger != null)
             {
-                _logger.Debug("Attempting to read: " + filename);
+                _logger.Debug("Attempting to read rule file: " + filename);
             }
 
+            if (!File.Exists(filename))
+                throw new FileNotFoundException();
+
             using (StreamReader file = File.OpenText(filename))
             {
                 AddString(file.ReadToEnd(), filename, tag);
             }
         }
 
         /// <summary>
-        /// Load rules from JSON string
+        ///     Adds the elements of the collection to the Ruleset
         /// </summary>
-        /// <param name="jsonstring">JSON string</param>
-        /// <param name="sourcename">Name of the source (file, stream, etc..)</param>
-        /// <param name="tag">Tag for the rules</param>
-        public void AddString(string jsonstring, string sourcename, string tag = null)
+        /// <param name="collection"> Collection of rules </param>
+        public void AddRange(IEnumerable<Rule> collection)
         {
-            List<Rule> ruleList;
-            JsonSerializerSettings settings = new JsonSerializerSettings()
+            foreach (var rule in collection.Select(AppInspectorRuleToOatRule))
             {
-                Error = HandleDeserializationError
-            };
+                if (rule != null)
+                {
+                    if (_logger != null)
+                    {
+                        _logger.Debug("Attempting to add rule: " + rule.Name);
```

Comment:
Replace this if statement and condition inside the statement with:

```suggestion
                        _logger?.Debug("Attempting to add rule: " + rule.Name);
```